app/controllers/application_controller.rb

```diff
@@ -885,10 +885,15 @@ def view_to_hash(view, fetch_data = false)
 
       next unless @row_button # Show a button in the last col
 
```

Comment:
This is not good :-(

This means adding code that is specific for a particular view to a very generic place.

Reply:
Yes, I agree that modifying a rather generic routine for one specific case probably isn't the best thing to do.

Could we perhaps decorate the `ServiceTemplate` class so that from the decorator we'd be driving the show / hide logic and the button popup titles as well?